1.regression_modelling/scripts/2.r2_figure.py

```diff
@@ -0,0 +1,657 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Calculate and generate R2 score figures to assess model performance across stains
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+
+import joblib
+import yaml
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+from sklearn.metrics import r2_score
+
+
+# ## Set paths
+
+# In[2]:
+
+
+# Output for figures
+figures_dir = pathlib.Path("./figures")
+figures_dir.mkdir(parents=True, exist_ok=True)
+
+
+# ## Define features per stain to use for filtering columns
+
+# In[3]:
+
+
+# Define the path to the YAML file with all the features per stain
+yaml_file_path = pathlib.Path("./features_dict.yml")
+
+# Load the column names dictionary from the YAML file
+with open(yaml_file_path, 'r') as yaml_file:
+    features_dict = yaml.safe_load(yaml_file)
+
+# Access the lists of features
+nucleus_features = features_dict['nucleus_features']
+a647_features = features_dict['a647_features']
+gold_features = features_dict['gold_features']
+
+# Now you can use these lists for further operations
+print(f"Nucleus Features: {len(nucleus_features)}")
+print(f"A647 Features: {len(a647_features)}")
+print(f"Gold Features: {len(gold_features)}")
```

Comment:
I like this import and feature organization style! No data wrangling every time. 

Are these feature selected features? Might be useful to include in the yaml as a comment?

Reply:
Thanks! :smile: These are the normalized features that are exported as a yml file in the previous notebook. I will make a code comment to mention it here.